zebedee-reader/src/main/java/com/github/onsdigital/zebedee/util/URIUtils.java

```diff
@@ -79,11 +83,93 @@ public static String getLastSegment(String uri) {
         return segments[segments.length - 1];
     }
 
-    public static String removeLastSegment(String uri) {
-        if (isEmpty(uri)) {
-            return null;
+    public static String removeAfterKeyword(String path, String keyword) {
+        int index = path.indexOf(keyword);
+        if (index == -1) {
+            return path;
+        }
+        String result = path.substring(0, index);
+        return result.endsWith("/") ? result.substring(0, result.length() - 1) : result;
+    }
+
+    public static String getQueryParameterFromURL(String url, String queryParam) {
+        try {
+            URI uri = new URI(StringUtils.lowerCase(url));
```

Comment:
Why is the URL being lowercased? URLs are case-sensitive, so this does not look right.

Reply:
👍 will remove

Reply:
url values and query param names are case sensitive